src/agent.c

```diff
@@ -547,6 +553,44 @@ int agent_add_remote_candidate(juice_agent_t *agent, const char *sdp) {
 	return ret;
 }
 
+int agent_add_turn_server(juice_agent_t *agent, juice_turn_server_t *turn_server) {
+	int ret = -1;
+	conn_lock(agent);
```

Comment:
`conn_lock()` has no effect before gathering as there is no `conn_impl` created at this point, you should not call it.